frame/ethereum/src/lib.rs

```diff
@@ -393,16 +394,17 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
-	pub fn transaction_weight(transaction_data: &TransactionData) -> (Option<Weight>, Option<u64>) {
-		match <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+	pub fn transaction_pov(transaction_data: &TransactionData) -> (Option<Weight>, Option<u64>){
+		let weight_limit = <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
 			transaction_data.gas_limit.unique_saturated_into(),
 			true,
-		) {
-			weight_limit if weight_limit.proof_size() > 0 => (
-				Some(weight_limit),
-				Some(transaction_data.proof_size_base_cost()),
-			),
-			_ => (None, None),
+		);
+		let proof_size_pre_execution = get_proof_size().unwrap_or_default();
```

Comment:
get_proof_size return the proof size consumed buy the whole block so far, it's not per transaction (expect at tax validation). So, tests with one transaction might pass but it will not work with several transactions

Reply:
This is the `proof_size` pre execution, used later to compute the PoV difference between `after` and `before` evm call. The base cost should no longer be necessary after https://github.com/paritytech/polkadot-sdk/pull/4765


Context: https://github.com/polkadot-evm/frontier/pull/1490#issuecomment-2283599749